apps/settings/js/panels/root/root.js

```diff
@@ -38,9 +40,65 @@ define(function(require) {
       });
     },
 
-    init: function root_init() {
-      // Load the necessary scripts after the UI update.
+    /**
+     * Update the sim related items based on mozMobileConnections.
+     */
+    _showSimItems: function root_showSimItems(panel) {
+      var mozMobileConnections = navigator.mozMobileConnections;
+      if (mozMobileConnections) {
+        if (mozMobileConnections.length === 1) { // single sim
+          var duelSimItem =
+            this._panel.querySelector('#simCardManager-settings');
+          duelSimItem.hidden = true;
+        } else { // DSDS
+          var simItem = this._panel.querySelector('#simSecurity-settings');
+          simItem.hidden = true;
+        }
+      } else {
+        // hide telephony panels
+        var elements = ['#call-settings',
+                        '#data-connectivity',
+                        '#messaging-settings',
+                        '#simSecurity-settings',
+                        '#simCardManager-settings'];
+        elements.forEach(el => {
+          this._panel.querySelector(el).hidden = true;
+        });
+      }
+    },
+    /**
+     * Update the developer menu item based on the preference.
+     */
+    _showDeveloperMenuItem: function root_showDeveloperMenuItem() {
+      var developerItem = this._panel.querySelector(
+        '[data-show-name="developer.menu.enabled"]');
+      if (developerItem && navigator.mozSettings) {
+        return navigator.mozSettings.createLock()
+          .get('developer.menu.enabled').then(
+            function(result) {
+              developerItem.hidden = !result['developer.menu.enabled'];
+          }, function(error) {
+            console.error(error);
+          });
+      } else {
+        return Promise.resolve();
+      }
+    },
+
+    init: function root_init(panel) {
+      this._panel = panel;
+      // Load the necessary scripts after the UI update
       setTimeout(this._loadScripts);
+
+      // Show NFC panel when support
+      var nfcItem = panel.querySelector('.nfc-settings');
+      nfcItem.hidden = !navigator.mozNfc;
+
+      // Show proper SIM panel
+      this._showSimItems();
+
+      // Show developer panel when necessary
+      this._showDeveloperMenuItem();
```

Comment:
The origin cause of RootPanelHandler in startup.js is to reduce the visible repaint at startup.

To delay the show/hide in root panel will cause extra screen reflow, it can be spotted by Toggle on `Developer>Flash Repainted Area`.